framework/base/BaseObject.php

```diff
@@ -150,6 +149,8 @@ public function __set($name, $value)
             $this->$setter($value);
         } elseif (method_exists($this, 'get' . $name)) {
             throw new InvalidCallException('Setting read-only property: ' . get_class($this) . '::' . $name);
+        } elseif ($this instanceof LooseSetInterface) {
```

Comment:
No need for `else` here. `if` on the next line is enough.

Reply:
See comment above.

Reply:
Actually I'm confused now. The current code already does not follow your ideas. The `else` here is not required:

```php
    public function __set($name, $value)
    {
        $setter = 'set' . $name;
        if (method_exists($this, $setter)) {
            $this->$setter($value);
        } elseif (method_exists($this, 'get' . $name)) {
            throw new InvalidCallException('Setting read-only property: ' . get_class($this) . '::' . $name);
        } else {
            throw new UnknownPropertyException('Setting unknown property: ' . get_class($this) . '::' . $name);
        }
    }
```

So please paste the code block the way you want it. I don't have time for more discussions.

Reply:
Ok, nevermind - it is required. TBH I hate wasting time on this.

Reply:
I give up at this point. If I remove the `elseif` as you want, then when a setter exists both, the setter and `$this->$name = $value` gets executed:

```php
$setter = 'set' . $name;
if (method_exists($this, $setter)) {
    $this->$setter($value);
} elseif (method_exists($this, 'get' . $name)) {
    throw new InvalidCallException('Setting read-only property: ' . get_class($this) . '::' . $name);
} 
if ($this instanceof LooseSetInterface) {
    $this->$name = $value;
} else {
    throw new UnknownPropertyException('Setting unknown property: ' . get_class($this) . '::' . $name);
}
```

Reply:
Ah, right. I've got confused. Leave it as is, it's not that important and I can adjust it later.